lib/src/components/list_item/notification_list_item.dart

```diff
@@ -0,0 +1,286 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+
+/// List Item for notifications
+class ZetaNotificationListItem extends StatefulWidget {
+  /// Constructor for [ZetaNotificationListItem]
+  const ZetaNotificationListItem({
+    super.key,
+    required this.notificationBadge,
+    required this.message,
+    required this.title,
+    this.notificationTime,
+    this.linkText,
+    this.linkOnClick,
+    required this.buttonOnClick,
+    this.hasMore = false,
+  });
+
+  /// Notification Badge to indicate type of notification or who it's coming from
+  final ZetaNotificationBadge notificationBadge;
+
+  /// Notification message
+  final String message;
+
+  /// Notification title
+  final String title;
+
+  /// Time of notificaiton
+  final String? notificationTime;
+
+  /// Attached link text
+  final String? linkText;
+
+  /// Link onClick method
+  final VoidCallback? linkOnClick;
+
+  /// User actions button on click
+  final VoidCallback buttonOnClick;
+
+  /// If notification is a grouped and there are more notifications
+  final bool? hasMore;
+
+  @override
+  State<ZetaNotificationListItem> createState() => _ZetaNotificationListItemState();
+
+  /// Function that returns copy of a notification item with altered fields
+  ZetaNotificationListItem copyWith(
+      {ZetaNotificationBadge? notificationBadge,
+      String? message,
+      String? title,
+      String? notificationTime,
+      String? linkText,
+      VoidCallback? linkOnClick,
+      VoidCallback? buttonOnClick,
+      bool? hasMore}) {
+    return ZetaNotificationListItem(
+      notificationBadge: notificationBadge ?? this.notificationBadge,
+      message: message ?? this.message,
+      title: title ?? this.title,
+      notificationTime: notificationTime ?? this.notificationTime,
+      linkText: linkText ?? this.linkText,
+      linkOnClick: linkOnClick ?? this.linkOnClick,
+      buttonOnClick: buttonOnClick ?? this.buttonOnClick,
+      hasMore: hasMore ?? this.hasMore,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(StringProperty('notificationTime', notificationTime))
+      ..add(StringProperty('message', message))
+      ..add(StringProperty('title', title))
+      ..add(StringProperty('linkText', linkText))
+      ..add(ObjectFlagProperty<VoidCallback?>.has('linkOnClick', linkOnClick))
+      ..add(ObjectFlagProperty<VoidCallback>.has('buttonOnClick', buttonOnClick))
+      ..add(DiagnosticsProperty<bool?>('hasMore', hasMore));
+  }
+}
+
+class _ZetaNotificationListItemState extends State<ZetaNotificationListItem> {
+  bool _notificationRead = false;
```

Comment:
This should be passed through as an argument, we should try to manage as little logic as possible from within these components.